unison-cli/src/Unison/CommandLine/InputPatterns.hs

```diff
@@ -3020,6 +3020,14 @@ patchArg =
       globTargets = Set.fromList []
     }
 
+upgradeArg :: ArgumentType
+upgradeArg =
+  ArgumentType
+    { typeName = "upgrade",
+      suggestions = \q cb _http p -> Codebase.runTransaction cb (prefixCompleteUpgrade q p),
+      globTargets = Set.fromList [Globbing.Term, Globbing.Type]
```

Comment:
We probably don't want `Globbing.Term` or `Globbing.Type` here, we are looking for the top-level paths under `lib`.